saleor/product/views.py

```diff
@@ -125,10 +125,34 @@ def category_index(request, path, category_id):
     now_sorted_by = get_now_sorted_by(product_filter)
     arg_sort_by = request.GET.get('sort_by')
     is_descending = arg_sort_by.startswith('-') if arg_sort_by else False
-    ctx = {'category': category, 'filter_set': product_filter,
-           'products': products_and_availability,
-           'products_paginated': products_paginated,
-           'sort_by_choices': SORT_BY_FIELDS,
-           'now_sorted_by': now_sorted_by,
-           'is_descending': is_descending}
+    ctx = {
+        'object': category, 'filter_set': product_filter,
+        'products': products_and_availability,
+        'products_paginated': products_paginated,
+        'sort_by_choices': SORT_BY_FIELDS,
+        'now_sorted_by': now_sorted_by,
+        'is_descending': is_descending}
     return TemplateResponse(request, 'category/index.html', ctx)
+
+
+def collection_index(request, pk):
+    collection = get_object_or_404(Collection, id=pk)
+    products = products_with_details(user=request.user).filter(
+        collections__id=collection.id).order_by('name')
+    product_filter = ProductCollectionFilter(
+        request.GET, queryset=products, collection=collection)
+    products_paginated = get_paginator_items(
+        product_filter.qs, settings.PAGINATE_BY, request.GET.get('page'))
+    products_and_availability = list(products_with_availability(
+        products_paginated, request.discounts, request.currency))
+    now_sorted_by = get_now_sorted_by(product_filter)
+    arg_sort_by = request.GET.get('sort_by')
+    is_descending = arg_sort_by.startswith('-') if arg_sort_by else False
+    ctx = {
```

Comment:
Since those views have so much in common, maybe we could extract a function `get_product_list_context(queryset, filter_set)` where:
- `queryset` - queryset of products from either a category or a collection
- `filter_set` - relevant, constructed filter object

We could call this function in both views to take care of constructing proper context which is almost the same in both cases.

Reply:
I made such function in `utils.py`.